flutter.el

```diff
@@ -129,13 +129,23 @@ ARGS is a space-delimited string of CLI flags passed to
    (let ((proc (get-buffer-process flutter-buffer-name)))
      (comint-send-string proc command))))
 
-(defun flutter--test (&optional args)
+(defun flutter--test (&rest args)
   "Execute `flutter test` inside Emacs.
 
-ARGS is a space-delimited string of CLI flags passed to
+ARGS is a list of CLI flags passed to
 `flutter`, and can be nil."
   (flutter--from-project-root
-   (compilation-start (format "%s test %s" (flutter-build-command) (or args "")) t)))
+   (compilation-start (format "%s test %s" (flutter-build-command) (mapconcat 'identity args " ")) t)))
+
+(defun flutter--find-test-case (line)
+  "Search backwards for test name starting at LINE on current buffer."
+  (save-excursion
+    (goto-char (point-min))
+    (forward-line (1- line))
+    (end-of-line)
+    (if (re-search-backward (concat "^[ \t]*\\(testWidgets\\|test\\|group\\)"
+                                    "(\\([\"'].*?[\"']\\),") nil t)
+        (match-string 2))))
```

Comment:
It's not really a big deal, but you can skip the extraneous first match by making the group ["shy"](https://www.gnu.org/software/emacs/manual/html_node/emacs/Regexp-Backslash.html#Regexp-Backslash) with `?:`. You can then do `(match-string 1)`.